build-common/src/org/jetbrains/kotlin/incremental/LookupStorage.kt

```diff
@@ -66,10 +67,27 @@ open class LookupStorage(
         }
     }
 
+    /** Set of [LookupSymbol]s that have been added or updated after the initialization of this [LookupStorage] instance. */
+    val addedOrUpdatedLookupSymbols: Set<LookupSymbolKey>
+        get() = run {
+            check(trackChanges) { "trackChanges is not enabled" }
+            trackedLookupMap.addedOrUpdatedKeys!!
+        }
+
+    /** Set of [LookupSymbol]s that have been removed after the initialization of this [LookupStorage] instance. */
+    val removedLookupSymbols: Set<LookupSymbolKey>
+        get() = run {
+            check(trackChanges) { "trackChanges is not enabled" }
+            trackedLookupMap.removedKeys!!
+        }
+
+    /** Returns all [LookupSymbol]s in this storage. Note that this call takes a bit of time to run. */
+    fun getLookupSymbolsExpensiveCall(): Collection<LookupSymbolKey> = trackedLookupMap.keys
```

Comment:
nit: I'd remove "ExpensiveCall" suffix from the name, it should be part of the javadoc. There are many methods that are non-trivial and we don't indicate that always in their name.

Reply:
Yes - Done